modifytags/modifytags.go

```diff
@@ -0,0 +1,367 @@
+		}
+
+		name = strings.Join(lowerSplitted, "_")
+	case "lispcase":
+		var lowerSplitted []string
+		for _, s := range splitted {
+			lowerSplitted = append(lowerSplitted, strings.ToLower(s))
+		}
+
+		name = strings.Join(lowerSplitted, "-")
+	case "camelcase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		titled[0] = strings.ToLower(titled[0])
+
+		name = strings.Join(titled, "")
+	case "pascalcase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, "")
+	case "titlecase":
+		var titled []string
+		for _, s := range splitted {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, " ")
+	case "keep":
+		name = fieldName
+	default:
+		unknown = true
+	}
+
+	if m.ValueFormat != "" {
+		prevName := name
+		name = strings.ReplaceAll(m.ValueFormat, "{field}", name)
+		if name == m.ValueFormat {
+			// support old style for backward compatibility
+			name = strings.ReplaceAll(m.ValueFormat, "$field", prevName)
+		}
+	}
+
+	for _, key := range m.Add {
+		splitted = strings.SplitN(key, ":", 2)
+		if len(splitted) >= 2 {
+			key = splitted[0]
+			name = strings.Join(splitted[1:], "")
+		} else if unknown {
+			// the user didn't pass any value but want to use an unknown
+			// transform. We don't return above in the default as the user
+			// might pass a value
+			return nil, fmt.Errorf("unknown transform option %q", m.Transform)
+		}
+
+		tag, err := tags.Get(key)
+		if err != nil {
+			// tag doesn't exist, create a new one
+			tag = &structtag.Tag{
+				Key:  key,
+				Name: name,
+			}
+		} else if m.Override {
+			tag.Name = name
+		}
+
+		if err := tags.Set(tag); err != nil {
+			return nil, err
+		}
+	}
+
+	return tags, nil
+}
+
+func isPublicName(name string) bool {
+	for _, c := range name {
+		return unicode.IsUpper(c)
+	}
+	return false
+}
+
+// Rewrite rewrites the node for structs between the start and end lines
+func (m *Mutation) Rewrite(fset *token.FileSet, node ast.Node, start, end int) (ast.Node, error) {
+	errs := &RewriteErrors{Errs: make([]error, 0)}
+
+	rewriteFunc := func(n ast.Node) bool {
+		x, ok := n.(*ast.StructType)
+		if !ok {
+			return true
+		}
+
+		for _, f := range x.Fields.List {
+			line := fset.Position(f.Pos()).Line
+
+			if !(start <= line && line <= end) {
+				continue
+			}
+
+			fieldName := ""
+			if len(f.Names) != 0 {
+				for _, field := range f.Names {
+					if !m.SkipUnexportedFields || isPublicName(field.Name) {
+						fieldName = field.Name
+						break
+					}
+				}
+			}
+
+			// anonymous field
+			if f.Names == nil {
+				ident, ok := f.Type.(*ast.Ident)
+				if !ok {
+					continue
+				}
+
+				if !m.SkipUnexportedFields {
+					fieldName = ident.Name
+				}
+			}
+
+			// nothing to process, continue with next line
+			if fieldName == "" {
+				continue
+			}
+
+			if f.Tag == nil {
+				f.Tag = &ast.BasicLit{}
```

Comment:
We're mutating here, so I don't think we need to return the node.

Therefore, I wonder if we even need to export the Rewrite API? Can we just call 'Apply' from main.go?